internal/resources/gitrepository/data_source_git_repository.go

```diff
@@ -0,0 +1,14 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package gitrepository
+
+import "github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+func DataSourceGitRepository() *schema.Resource {
```

Comment:
What is the purpose of this datasource? Is it really required, only the schema is defined here

Reply:
I've rebased it, so git repository resource is not there in this MR